python/ray/data/tests/test_execution_optimizer.py

```diff
@@ -1334,20 +1335,15 @@ def test_from_huggingface_e2e(ray_start_regular_shared, enable_optimizer):
         assert len(ds.take_all()) > 0
         # Check that metadata fetch is included in stats;
         # the underlying implementation uses the `FromArrow` operator.
-        assert "FromArrow" in ds.stats()
-        assert ds._plan._logical_plan.dag.name == "FromArrow"
-        assert ray.get(ray_datasets[ds_key].to_arrow_refs())[0].equals(
-            data[ds_key].data.table
-        )
-        _check_usage_record(["FromArrow"])
-
-    ray_dataset = ray.data.from_huggingface(data["train"])
-    assert isinstance(ray_dataset, ray.data.Dataset)
-    assert len(ray_dataset.take_all()) > 0
-    assert "FromArrow" in ray_dataset.stats()
-    assert ray_dataset._plan._logical_plan.dag.name == "FromArrow"
-    assert ray.get(ray_dataset.to_arrow_refs())[0].equals(data["train"].data.table)
-    _check_usage_record(["FromArrow"])
+        # assert "FromArrow" in ds.stats()
+        assert ds._plan._logical_plan.dag.name == "ReadParquet"
```

Comment:
can we add a comment briefly explaining why we check for `ReadParquet` in this test for Hugging Face?

or even better would be to have two test paths, one which uses the new distributed parquet path, and one which uses the backup direct read from HF dataset. I know that might be a bit tricky though, since you'll likely need to find a private dataset (that we can access) or a public dataset that doesn't have parquet files converted for some reason. So no worries if you cannot find a way to easily test both paths

Reply:
Ended up adding a second test case which tests the same on a modified dataset (just did a train test split and had similar tests as the unmodified), so we should be testing both paths now!